File: client/src/app/classes/use-hint-game-replay-command/use-hint-game-replay-command.ts
import { GameReplayCommand } from '@app/interfaces/game-replay-command';
import { ClassicModeService } from '@app/services/classic-mode/classic-mode.service';
import { ReplayEventDto } from '@common/model/events/replay.events';
import { Hint } from '@common/model/hints';

export class UseHintGameReplayCommand implements GameReplayCommand {
    time: number;
    private classicModeService: ClassicModeService;
    private replayEventDto: ReplayEventDto;

    constructor(classicModeService: ClassicModeService, replayEventDto: ReplayEventDto) {
        this.time = replayEventDto.time;
        this.classicModeService = classicModeService;
        this.replayEventDto = replayEventDto;
    }

    action() {
        this.classicModeService.hintCallback(this.replayEventDto.response as Hint);
    }
}

File: client_leger/lib/classes/game-replay-commands/use_hint_game_replay_command.dart
import 'package:client_leger/classes/game-replay-commands/game_replay_command.dart';
import 'package:client_leger/interfaces/dto/replay_event.dto.dart';

class UseHintGameReplayCommand extends GameReplayCommand {
  final UseHintReplayDto eventDto;

  UseHintGameReplayCommand({required this.eventDto, required super.time});

  @override
  void action() {
    print('UseHintGameReplayCommand action');
  }
}
